app/src/main/java/net/gsantner/markor/util/AppSettings.java

```diff
@@ -339,78 +341,98 @@ public void toggleFavouriteFile(File file) {
     private static final String PREF_PREFIX_PREVIEW_STATE = "PREF_PREFIX_PREVIEW_STATE";
     private static final String PREF_PREFIX_INDENT_SIZE = "PREF_PREFIX_INDENT_SIZE";
 
-    public void setLastEditPosition(File file, int pos, int scrolloffset) {
-        if (file == null || !file.exists()) {
-            return;
-        }
-        if (!file.equals(getTodoFile()) && !file.equals(getQuickNoteFile())) {
-            setInt(PREF_PREFIX_EDIT_POS_CHAR + file.getAbsolutePath(), pos, _prefCache);
-            setInt(PREF_PREFIX_EDIT_POS_SCROLL + file.getAbsolutePath(), scrolloffset, _prefCache);
+    private boolean pathOk(final String path) {
+        return (path != null && !path.trim().isEmpty());
+    }
+
+    public void setLastEditPosition(final String path, int pos, int scrolloffset) {
+        // Includes hack to prevent last edit from being set for quicknote and todo
+        if (pathOk(path)) {
```

Comment:
Removed the special case code here and using paths, not `File` objects. Special case logic now handled in the `DocumentEditFragment` directly.